janus-gateway/Dockerfile

```diff
@@ -1,57 +1,68 @@
-FROM ubuntu:20.04
+FROM ubuntu:24.04
 
-ENV DEBIAN_FRONTEND noninteractive
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y tzdata
+
+ENV PATH="/root/.local/bin:$PATH"
+
+ENV PATH="/root/.local/bin:$PATH"
 
 # Install Dependencies
-RUN apt-get update -y && apt-get install -y \
-    git wget cmake \
+RUN apt-get install -y \
+    git wget cmake build-essential \
     libmicrohttpd-dev libjansson-dev \
     libssl-dev libsofia-sip-ua-dev libglib2.0-dev \
     libopus-dev libogg-dev libcurl4-openssl-dev liblua5.3-dev \
     libconfig-dev pkg-config gengetopt libtool automake \
-    python3-pip
+    pipx && \
+    rm -rf /var/lib/apt/lists/*
 
 # Install libnice
-RUN pip3 install meson ninja && \
+RUN pipx install meson ninja && \
     git clone https://gitlab.freedesktop.org/libnice/libnice ~/libnice && \
     cd ~/libnice && \
-    meson --prefix=/usr build && ninja -C build && ninja -C build install
+    meson --prefix=/usr build && ninja -C build && ninja -C build install && \
+    rm -rf ~/libnice
 
 # Install libsrtp
 RUN cd ~ && \
     wget https://github.com/cisco/libsrtp/archive/v2.2.0.tar.gz && \
     tar xfv v2.2.0.tar.gz && \
     cd libsrtp-2.2.0 && \
     ./configure --prefix=/usr --enable-openssl && \
-    make shared_library && make install
+    make shared_library && make install && \
+    rm -rf ~/libsrtp-2.2.0 v2.2.0.tar.gz
 
 # Install libwebsockets
-RUN git clone https://libwebsockets.org/repo/libwebsockets ~/libwebsockets && \
+RUN git clone https://github.com/warmcat/libwebsockets.git ~/libwebsockets && \
```

Comment:
The repository was not available for now.
I don't know if it's temporary.

I replaced url as following documentation.

> * *Note:* if libwebsockets.org is unreachable for any reason, replace the first line with this:
> 
> 	git clone https://github.com/warmcat/libwebsockets.git
https://janus.conf.meetecho.com/docs/README.html